src/Processors/Executors/PullingAsyncPipelineExecutor.cpp

```diff
@@ -134,7 +134,8 @@ bool PullingAsyncPipelineExecutor::pull(Chunk & chunk, uint64_t milliseconds)
     {
         chunk = lazy_format->getChunk(milliseconds);
         data->rethrowExceptionIfHas();
-        return true;
+        bool is_finished = lazy_format->isFinished() && !chunk;
```

Comment:
There is no way we get here in case `lazy_format && lazy_format->isFinished()` because previous if clause will `return false`

Reply:
Just looks like this code can be simplified or is it better to leave it like this?

Reply:
As I get `lazy_format->isFinished` may be updated in a background, maybe while `getChunk`

Reply:
Oh, okay then